File: backend/src/main/java/com/frutiva/backend/modelos/producto/Producto.java
package com.frutiva.backend.modelos.producto;

import java.time.LocalDate;

import com.frutiva.backend.modelos.usuario.Usuario;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Producto {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String sku;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false)
    private String marca;

    @Column
    private String modelo;

    @Column
    private String categoria;

    @ManyToOne
    @JoinColumn(name = "vendedor", nullable = false)
    private Usuario vendedor;

    @Column(nullable = false)
    private Long stock;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column
    private double pesoNeto;    

    @Column
    private double pesoBruto;   

    @Column
    private double largo;       

    @Column
    private double ancho;     

    @Column
    private double alto;    

    @Column
    private String color;

    @Column
    private String material;

    @Column
    private LocalDate fechaFabricacion;

    @Column
    private LocalDate fechaCaducidad;

    @Column
    private String origen;       

    @Column
    private String proveedor;      
    
    @Column
    private Integer garantia;    
    
    @Column(nullable = false)
    private double precioUnitario;

    @Column(nullable = false)
    private int cantidadMinima;

    @Column(nullable = false)
    private double precioPorMayor;

    @Column
    private Integer unidadesPorPaquete;  

    @Column
    private String tipoEmpaque;  
    @Column
    private String numeroLote;       

    @Column
    private Integer descuentoPorcentaje; 

}

File: frontend/src/app/models/Producto.ts
import { Usuario } from "./Usuario";

export class Producto {
    constructor(
      public id?: number | null,
      public sku?: string | null,
      public nombre?: string | null,
      public marca?: string | null,
      public modelo?: string | null,
      public categoria?: string | null,
      public vendedor?: Usuario | null,
      public stock?: number | null,
      public descripcion?: string | null,
      public pesoNeto?: number | null,
      public pesoBruto?: number | null,
      public largo?: number | null,
      public ancho?: number | null,
      public alto?: number | null,
      public color?: string | null,
      public material?: string | null,
      public fechaFabricacion?: Date | null, 
      public fechaCaducidad?: Date | null,
      public origen?: string | null,
      public proveedor?: string | null,
      public garantia?: number | null,
      public precioUnitario?: number | null,
      public cantidadMinima?: number | null,
      public precioPorMayor?: number | null,
      public unidadesPorPaquete?: number | null,
      public tipoEmpaque?: string | null,
      public numeroLote?: string | null,
      public descuentoPorcentaje?: number | null
    ) {}
  }
  